spring-integration-core/src/test/java/org/springframework/integration/dispatcher/FailingCommitTransactionManager.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2002-2008 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.dispatcher;
+
+import org.springframework.integration.util.TestTransactionManager;
+import org.springframework.transaction.IllegalTransactionStateException;
+import org.springframework.transaction.TransactionException;
+import org.springframework.transaction.support.DefaultTransactionStatus;
+
+/**
+ * @author Andreas Baer
+ */
+@SuppressWarnings("serial")
+public class FailingCommitTransactionManager extends TestTransactionManager {
```

Comment:
M-m-m. This looks like a test-case scope.
Let's move it into the test class - `PollingTransactionTests`!